__tests__/app.test.js

```diff
@@ -178,4 +178,75 @@ describe('4. GET /api/articles/:article_id/comments', () => {
                 })
         })
     })
+})
+
+describe('5. POST /api/articles/:article_id/comments', () => {
+    describe('a. status 201 & data', () => {
+        it('returns 201 status code and the object of the comment back to the user', () => {
+            return db.query("INSERT INTO users (username, name) VALUES ('jxnbox', 'jan');")
+            .then ( () => {
+                const newCommentOnj = {username: "jxnbox", body: "amazing! 10/10"}
+    
+                return request(app)
+                .post('/api/articles/7/comments')
+                .send(newCommentOnj)
+                .expect(201)
+                .then( (res) => {
+                    const {comment} = res.body;
+                    expect(comment).toEqual("amazing! 10/10");
+                })
+            })
+        })
+
+    })
+
+    describe('b. error handling', () => {
+        it('return 400 status code when the wrong datatype is given for the ID', () => {
+            return db.query("INSERT INTO users (username, name) VALUES ('jxnbox', 'jan');")
+            .then ( () => {
+                const newCommentOnj = {username: "jxnbox", body: "amazing! 10/10"}
+
+                return request(app)
+                .post('/api/articles/article7/comments')
+                .send(newCommentOnj)
+                .expect(400)
+                .then( (res) => {
+                    const {msg} = res.body
+                    expect(msg).toBe("Bad Request")
+                })
+            })
+        })
+
+        it('return 404 status code when a valid ID datatype is given but ID does not exist', () => {
+            return db.query("INSERT INTO users (username, name) VALUES ('jxnbox', 'jan');")
+            .then ( () => {
+                const newCommentOnj = {username: "jxnbox", body: "amazing! 10/10"}
+
+                return request(app)
+                .post('/api/articles/999/comments')
+                .send(newCommentOnj)
+                .expect(404)
+                .then( (res) => {
+                    const {msg} = res.body
+                    expect(msg).toBe('Not Found')
+                })
+            })
+        })
+
+        it('return 400 status code when a valid ID datatype is given but the Object does not contain a username', () => {
+            return db.query("INSERT INTO users (username, name) VALUES ('jxnbox', 'jan');")
+            .then ( () => {
+                const newCommentOnj = { body: "amazing! 10/10"}
+
+                return request(app)
+                .post('/api/articles/7/comments')
+                .send(newCommentOnj)
+                .expect(400)
+                .then( (res) => {
+                    const {msg} = res.body
+                    expect(msg).toBe('Bad Request')
+                })
+            })
+        })
+    })

```

Comment:
Missing test - 404 - What happens when I make a request using a username that isn't found in the database? eg. {username: "Rob", body: "I don't have an account"}